repository/src/main/java/org/apache/atlas/ESAliasStore.java

```diff
@@ -0,0 +1,270 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.atlas;
+
+import org.apache.atlas.ESAliasRequestBuilder.AliasAction;
+import org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil;
+import org.apache.atlas.accesscontrol.persona.PersonaContext;
+import org.apache.atlas.accesscontrol.purpose.AtlasPurposeUtil;
+import org.apache.atlas.accesscontrol.purpose.PurposeContext;
+import org.apache.atlas.exception.AtlasBaseException;
+import org.apache.atlas.model.instance.AtlasEntity;
+import org.apache.atlas.repository.graphdb.AtlasGraph;
+import org.apache.atlas.repository.store.graph.v2.EntityGraphRetriever;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.atlas.ESAliasRequestBuilder.ESAliasAction.ADD;
+import static org.apache.atlas.accesscontrol.AccessControlUtil.ACCESS_ENTITY_READ;
+import static org.apache.atlas.accesscontrol.AccessControlUtil.getActions;
+import static org.apache.atlas.accesscontrol.AccessControlUtil.getESAliasName;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.getAssets;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.getConnectionId;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.getIsAllow;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.getQualifiedName;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.mapOf;
+import static org.apache.atlas.accesscontrol.purpose.AtlasPurposeUtil.getTags;
+import static org.apache.atlas.repository.Constants.PROPAGATED_TRAIT_NAMES_PROPERTY_KEY;
+import static org.apache.atlas.repository.Constants.QUALIFIED_NAME;
+import static org.apache.atlas.repository.Constants.TRAIT_NAMES_PROPERTY_KEY;
+import static org.apache.atlas.repository.Constants.VERTEX_INDEX_NAME;
+
+@Component
+public class ESAliasStore implements IndexAliasStore {
+    private static final Logger LOG = LoggerFactory.getLogger(ESAliasStore.class);
+
+    private final AtlasGraph graph;
+    private final EntityGraphRetriever entityRetriever;
+
+    @Inject
+    public ESAliasStore(AtlasGraph graph,
```

Comment:
Constructor injection can be called easily with lombok's @RequiredArgsConstructor